app/src/main/java/com/android/sample/model/recipe/RecipesViewModel.kt

```diff
@@ -325,7 +328,12 @@ class RecipesViewModel(
         onSuccess = { uri ->
           val newRecipe = recipe.copy(url = uri)
           repository.addDownload(
-              newRecipe, onSuccess = { onSuccess(newRecipe) }, onFailure = { e -> onFailure(e) })
+              newRecipe,
+              onSuccess = {
+                _recipesDownload.value += newRecipe
```

Comment:
I know it's a bit of a nitpick, but I think we should refactor the name of `recipesDowload` to `downloadedRecipes`, it's clearer.

Reply:
Done!